packages/trpc-server/src/routers/me-router.ts

```diff
@@ -13,16 +14,110 @@ export const meRouter = router({
       return result
     }),
 
+  trackRelationship: publicProcedure
+    .input(z.string())
+    .query(async ({ ctx, input }) => {
+      return ctx.loaders.trackRelationLoader.load(parseInt(input))
+    }),
+
+  playHistory: publicProcedure
+    .meta({ openapi: { method: 'GET', path: '/me/play_history' } })
+    .input(
+      z.object({
+        sort: z
+          .enum([
+            'trackName',
+            'artistName',
+            'releaseDate',
+            'playDate',
+            'duration',
+            'playCount',
+            'repostCount',
+          ])
+          .default('playDate'),
+        sortAscending: z.boolean().default(false),
+        cursor: z.number().default(0),
+        limit: z.number().default(1000),
+      })
+    )
+    .output(
+      z.array(
+        z.object({
+          artistId: z.number(),
+          artistHandle: z.string(),
+          artistName: z.string(),
+          trackId: z.number(),
+          routeId: z.string().nullish(),
+          trackName: z.string(),
+          releaseDate: z.date(),
+          duration: z.number(),
+          playCount: z.number(),
+          repostCount: z.number(),
+          playDate: z.date(),
+        })
+      )
+    )
+    .query(async ({ ctx, input }) => {
+      if (!ctx.currentUserId) {
+        throw new TRPCError({ code: 'UNAUTHORIZED' })
+      }
+      const sortMapping: Record<typeof input.sort, any> = {
+        trackName: sql`track_name`,
+        artistName: sql`artist_name`,
+        releaseDate: sql`release_date`,
+        duration: sql`duration`,
+        playDate: sql`play_date`,
+        playCount: sql`play_count`,
+        repostCount: sql`repost_count`,
+      }
+      const sortField = sortMapping[input.sort]
+      const sortDirection = input.sortAscending ? sql`asc` : sql`desc`
+
+      return sql`
+        with duped as (
+
+          select
+            a.user_id as artist_id,
+            a.handle as artist_handle,
+            a.name as artist_name,
+
+            t.track_id,
+            t.route_id,
+            t.title as track_name,
+            coalesce(t.release_date, t.created_at) as release_date,
+            t.duration,
+
+            agg_play.count as play_count,
+            agg_track.repost_count,
+            p.created_at as play_date,
+            row_number() over (partition by p.play_item_id order by p.created_at asc) as rownum
+
+          from plays p
+          join tracks t on p.play_item_id = t.track_id
+          join users a on t.owner_id = a.user_id
+          join aggregate_plays agg_play on p.play_item_id = agg_play.play_item_id
+          join aggregate_track agg_track on p.play_item_id = agg_track.track_id
+
+          where p.user_id = ${ctx.currentUserId!}
```

Comment:
I think long term this should use a wallet address recovered from a signature, using a timestamp as replay prevention, similar to how we did for DMs. On DN right now a number of endpoints follow this pattern (though without the timestamp check nor the correct data in the signature data).